### PR TITLE
Normalise changelog headings

### LINKED ISSUE #44: Handle CHANGELOGs with "v" in the headings
### 🤔 What's the problem you're trying to solve?

We're not consistent about whether we use a "v" in the `h2`s in our CHANGELOGs

Example: https://github.com/cucumber/html-formatter/blob/main/CHANGELOG.md#1910---2022-04-15
Example: https://github.com/cucumber/multi_test/blob/main/CHANGELOG.md#v012

We should handle that, instead of doing this:

```
polyglot-release 1.0.0
No git tag found for vv0.1.2 (found in CHANGELOG.md)!

Do you need to run this?
    git fetch --tags
```
### ✨ What's your proposed solution?

I would like us to normalise the headings, only adding a prefix "v" if it's not already there. So we can handle both styles.

### ⛏ Have you considered any alternatives or workarounds?

Throw an error if the heading starts with a "v" and ask the user to clean up their CHANGELOG.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/ISSUE_TEMPLATE/feature_request.md)*

### PR DESCRIPTION
**Pull request by mattwynne**
### 🤔 What's changed?

* Refined the check that compares the `changelog latest` with the local tags (to ensure total tags have been fetched) so it can handle whether or not the `CHANGELOG.md` has `v` prefixes on the version headings.

### ⚡️ What's your motivation? 

We have some variance in whether our repos have this `v` prefix and it seems better to handle that than go out and tidy them all up.

Fixes #44

### 🏷️ What kind of change is this?

- :zap: New feature (non-breaking change which adds new behaviour)

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [x] I've changed the behaviour of the code
  - [x] I have added/updated tests to cover my changes.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

File: polyglot-release [hunk 1]
```diff
@@ -182,8 +182,14 @@ function check_in_git_root_directory() {
 }
 
 function check_git_tags_fetched() {
-  if [[ -z "$(git tag --list "v$(changelog latest)")" ]]; then
-    echo "No git tag found for v$(changelog latest) (found in CHANGELOG.md)!"
+  latest_version_from_changelog=$(changelog latest)
+  if [[ $latest_version_from_changelog == v* ]]; then
+    expected_tag=$latest_version_from_changelog
+  else
+    expected_tag=v$latest_version_from_changelog
```
Comment (mpkorstanje) [inline polyglot-release:189]: Consider both strings for consistency.

File: tests/fixtures/ruby/CHANGELOG.md [hunk 2]
```diff
@@ -10,9 +10,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 ### Fixed
 - This is a test
 
-## [0.0.1] - 2000-01-01
+## [v0.0.1] - 2000-01-01
 
-## [0.0.0] - 2000-01-01
+## [v0.0.0] - 2000-01-01
```
Comment (mpkorstanje) [inline tests/fixtures/ruby/CHANGELOG.md:15]: This now looks like a test for two things.

**Review by mpkorstanje — APPROVED** [review]

File: polyglot-release [hunk 3]
```diff
@@ -182,8 +182,14 @@ function check_in_git_root_directory() {
...
-  if [[ -z "$(git tag --list "v$(changelog latest)")" ]]; then
-    echo "No git tag found for v$(changelog latest) (found in CHANGELOG.md)!"
+  latest_version_from_changelog=$(changelog latest)
+  if [[ $latest_version_from_changelog == v* ]]; then
+    expected_tag="$latest_version_from_changelog"
+  else
+    expected_tag="v$latest_version_from_changelog"
+  fi
+  if [[ -z $(git tag --list "$expected_tag") ]]; then
+    echo "No git tag found for $($expected_tag) (found in CHANGELOG.md)!"
```
Comment (mpkorstanje) [inline polyglot-release:192]: I think the subshell here is a mistake.

**Comment by mattwynne** [comment]
Sorry, I should have made this Draft, I was over-confident and didn't run the tests locally! 😊 

I've reverted to draft now. Let me tidy it up before you look over it again.

**Review by mpkorstanje — APPROVED** [review]

**Pull request merged**